### DISCUSSION TITLE
What are the advantages of this over Kamal 2?

### DISCUSSION THREAD

**Discussion by gap777** [MightyMoud/sidekick#39] [General]
Ruby on Rails / 37 signals / DHH just announced Kamal 2 that seems very similar to this.
Can you help us understand the differences between this package and the new Kamal 2?
Thanks!

**Comment by MightyMoud** [comment]
Hey @gap777,

Thanks for asking this question. I wanted to add a section to readme titled "What about X" but didn't get the time.

Kamal 2 is a great tool and it works super well. I think with Kamal 1 it didn't support multi apps per server, while Kamal 2 does making it very similar to Sidekick. Kamal however has some certain differences in approach:
- Kamal still needs a registry to push images to
- Kamal is based on config rather than interactive inputs on terminal

I think that is it. Kamal is far more developed than Sidekick tho. If it solves your problem you should probably stick with it. 
In my ultimate vision for Sidekick it will do server monitoring, logs, migrations and db backups. Not sure how Kamal stands on those. 
Yet again if Kamal solves your problem I think you should use it at least till Sidekick is more mature.

Cheers

**Comment by rlnorthcutt** [comment]
I think another advantage (for _some_ people) is that Sidekick is written in Go. Kamal is written in Ruby. Nothing wrong with Ruby at all, but Go will likely be more performant overall. This may not matter much for the orchestration layer, but it is still a consideration, especially with future plans like monitoring, logs, etc. Also, for a Go developer, this is likely to be a big plus.

However, as @MightyMoud points out, Kamal is more mature and has the weight of DHH/37Signals, so it may be more desirable for those who are more interested in stability and community size.
